### ISSUE TITLE
Themes with embedded images (like Agency) - ressource paths used in css url() should not be case sensitive in live environment 

### ISSUE THREAD

**Issue by bashuss** [OrchardCMS/OrchardCore#9104]
### Is your feature request related to a problem? Please describe.
When working on images and css of a theme with embedded images, the development environment behaves differently regarding case sensitivity than the live site (even on windows server). This can create frustration and long searching for problem.
e.g. your image automatically gets an uppercase .JPG by your editing software, but you are used to using everything lower case in css. Your image will be found in development with VS on Windows, but not on the live server after deployment. Firefox will give you the message, that the image is corrupted (instead of "not found").
Yes, this is nothing new for people working with case sensitive file systems, but a .NET CMS will still attract many Windows centered developers and users.

### Describe the solution you'd like
- Make access to embedded ressources by web requests case insensitive

### Describe alternatives you've considered
- Make access in development environment consistent to live environment (e.g. case sensitive)
- Create usefull errors for log and browser, if ressources are not found
- Create build warnings, if embedded ressources are not fully lower case

**Comment by Skrypt** [comment]
You can use Media Slugify feature in OC to mitigate these issues.

![image](https://user-images.githubusercontent.com/3228637/114225216-0f627700-9940-11eb-8036-522778fa6ae8.png)

**Comment by bashuss** [comment]
No, you do not really understand my issue. I am aware of this Linux server behavior. But I am hosting on a Windows Server.
The issue appears, as during development in VisualStudio Files in a Theme like "The Agency" are hosted directly from the wwwroot folder. You can throw in, change and rename your files at any time.
But after deployment, on the hosting server, there are no such folders for the Agency Theme plugin, yet all the files are reachable by URL. In fact all the Theme files have been embedded into the Theme Assembly are served from there. This access to the Theme Assembly Ressources seems to be case sensitive. I think this is just a matter of decision, as this is most likely handled by orchard code - by I have not checked. I might be also part of the framework.

**Comment by Skrypt** [comment]
Ok, maybe there is something there but I'd rather keep the default behavior of a MVC project if that is.

/cc @jtkech

**Comment by jtkech** [comment]
Yes, for embedded files we are case sensitive to prevent conflicts with other files on a case sensitive file system.

But yes, for now on windows a physical file provider is always case insensitive, so idem for the part of a static file provider using physical file providers, they plan to implement a case sensitive option but not yet done.

**Comment by Skrypt** [comment]
So basically we follow what MVC does... nothing to fix here.

**Comment by jtkech** [comment]
While in dev we could make e.g. our `ModuleProjectStaticFileProvider` case sensitive, but for now we can't retrieve the exact file name from a file info, see https://github.com/dotnet/runtime/issues/14321
